gamestate.js

```diff
@@ -123,28 +123,23 @@ function updateValue(elt, newValue, toggleOnZero) {
 }
 
 // Update life totals, with debouncing
-var updateP1Numbers = _.debounce(function(v) {
-    updateValue($(".p1.life"), v.val().life);
-    updateValue($(".p1.poison"), v.val().poison, true);
+function updateSide(prefix, v) {
+    updateValue($(prefix + ".life"), v.val().life);
+    updateValue($(prefix + ".poison"), v.val().poison, true);
     if (v.val().gamewins > 0) {
-        updateValue($(".p1.wins"), v.val().gamewins);
+        updateValue($(prefix + ".wins"), v.val().gamewins);
     } else {
-        $(".p1.wins").text(v.val().gamewins);
+        $(prefix + ".wins").text(v.val().gamewins);
     }
-}, 1000);
+}
 
-var updateP2Numbers = _.debounce(function(v) {
-    updateValue($(".p2.life"), v.val().life);
-    updateValue($(".p2.poison"), v.val().poison, true);
-    if (v.val().gamewins > 0) {
-        updateValue($(".p2.wins"), v.val().gamewins);
-    } else {
-        $(".p2.wins").text(v.val().gamewins);
-    }
+var updateAllNumbers = _.debounce(function () {
+    firebase.database().ref('player1').once('value', updateSide.bind(null, ".p1"));
```

Comment:
I tested this, and it works.  (Live at http://verylowsodium.com/urza/overlay-game.html).  What a _debounce instance does, when called, is save the arguments (none here of course), and start a timer.  If it's called again before the timer expires, the timer is restarted and the new arguments are saved.

When the timer finally expires, the wrapped object is called with the latest saved arguments.

So in this case, when things have stopped changing after 1000ms, we fetch the current values of 'player1' and 'player2' and do all updates.